tests/Neo.UnitTests/Cryptography/UT_Crypto.cs

```diff
@@ -94,6 +97,133 @@ public void TestSecp256k1()
+            var signature2 = "4e0ea79d4a476276e4b067facdec7460d2c98c8a65326a6e5c998fd7c65061140e45aea5034af973410e65cf97651b3f2b976e3fc79c6a93065ed7cb69a2ab5a01".HexToBytes();
+            var expectedPubKey2 = "02dbf1f4092deb3cfd4246b2011f7b24840bc5dbedae02f28471ce5b3bfbf06e71".HexToBytes();
+
+            var recoveredKey2 = Crypto.ECRecover(signature2, messageHash2);
+            recoveredKey2.EncodePoint(true).Should().BeEquivalentTo(expectedPubKey2);
+
+            // Test case 3 - recovery param 0
+            var message3 = "db0d31717b04802adbbae1997487da8773440923c09b869e12a57c36dda34af11b8897f266cd81c02a762c6b74ea6aaf45aaa3c52867eb8f270f5092a36b498f88b65b2ebda24afe675da6f25379d1e194d093e7a2f66e450568dbdffebff97c4597a00c96a5be9ba26deefcca8761c1354429622c8db269d6a0ec0cc7a8585c".HexToBytes();
+            var messageHash3 = "c36d0ecf4bfd178835c97aae7585f6a87de7dfa23cc927944f99a8d60feff68b".HexToBytes();
+            var signature3 = "f25b86e1d8a11d72475b3ed273b0781c7d7f6f9e1dae0dd5d3ee9b84f3fab89163d9c4e1391de077244583e9a6e3d8e8e1f236a3bf5963735353b93b1a3ba93500".HexToBytes();
+            var expectedPubKey3 = "03414549fd05bfb7803ae507ff86b99becd36f8d66037a7f5ba612792841d42eb9".HexToBytes();
+
+            var recoveredKey3 = Crypto.ECRecover(signature3, messageHash3);
+            recoveredKey3.EncodePoint(true).Should().BeEquivalentTo(expectedPubKey3);
+
+            // Test invalid cases
+            var invalidSignature = new byte[65];
+            Action action = () => Crypto.ECRecover(invalidSignature, messageHash1);
+            action.Should().Throw<ArgumentException>();
+
+            // Test with invalid recovery value
+            var invalidRecoverySignature = signature1.ToArray();
+            invalidRecoverySignature[64] = 29; // Invalid recovery value
+            action = () => Crypto.ECRecover(invalidRecoverySignature, messageHash1);
+            action.Should().Throw<ArgumentException>();
+
+            // Test with wrong message hash
+            var recoveredWrongHash = Crypto.ECRecover(signature1, messageHash2);
+            recoveredWrongHash.EncodePoint(true).Should().NotBeEquivalentTo(expectedPubKey1);
+        }
+
+        [TestMethod]
+        public void TestERC2098()
+        {
+            //
+            // Private Key: 0x1234567890123456789012345678901234567890123456789012345678901234
+            // Message: "It's a small(er) world"
+            // Signature:
+            // r:  0x9328da16089fcba9bececa81663203989f2df5fe1faa6291a45381c81bd17f76
+            // s:  0x139c6d6b623b42da56557e5e734a43dc83345ddfadec52cbe24d0cc64f550793
+            // v:  28
+
+
+            // Test from https://eips.ethereum.org/EIPS/eip-2098
+            var privateKey = "1234567890123456789012345678901234567890123456789012345678901234".HexToBytes();
+
+            var expectedPubKey1 = (Neo.Cryptography.ECC.ECCurve.Secp256k1.G * privateKey).ToArray();
```

Comment:
Expected values 
pubKeyX - `0xe90c7d3640a1568839c31b70a893ab6714ef8415b9de90cedfc1c8f353a6983e`
pubKeyY - `0x625529392df7fa514bdd65a2003f6619567d79bee89830e63e932dbd42362d34`

eth address for reference - `0x2e988A386a799F506693793c6A5AF6B54dfAaBfB`